.github/workflows/publish.yml

```diff
@@ -0,0 +1,53 @@
+name: Publish to PyPI
+on:
+  push:
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+
+jobs:
+  publish-release:
+    name: Publish release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name:
+        run: |
+          package_version=$(grep 'version' setup.py | sed -E 's|.*([0-9]+\.[0-9]+\.[0-9]+).*|\1|')
+          tag_version=$(echo "$GITHUB_REF" | sed -E 's|.*([0-9]+\.[0-9]+\.[0-9]+).*|\1|')
+          if [ "$package_version" != "$tag_version" ]; then
+            echo "Tag version (${tag_version}) does not match package version (${package_version})"
+            exit 1
+          fi
+
+          if ! grep -iq "version ${package_version}" CHANGELOG.md; then
+            echo "Missing changelog section for version ${package_version}"
+            exit 1
+          fi
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7
```

Comment:
Just a curiosity question, will there become a time that we want to update this python version?

Reply:
I copied some of these steps from the PR checks workflow, which uses Python 3.7. Both workflows should probably be updated to use Python 3.8, since that is the version on Dataproc clusters.
https://cloud.google.com/dataproc/docs/tutorials/python-configuration#image_python_environments